subsys/bluetooth/services/ots/ots.c

```diff
@@ -351,6 +388,13 @@ int bt_ots_obj_delete(struct bt_ots *ots, uint64_t id)
 		return -EBUSY;
 	}
 
+	if (ots->cb->obj_deleted) {
```

Comment:
I see that this is currently also missing from other checks, but I guess we should probably check `ots->cb == NULL` before we check for the individual callback functions, even if we have the __ASSERT in the init function, as the upper layer can potential remove this at any time.